server/blob/blob-s3/src/main/java/org/apache/james/blob/objectstorage/aws/S3BlobStoreDAO.java

```diff
@@ -159,6 +162,7 @@ public long size() {
             .endpointOverride(authConfiguration.getEndpoint())
             .region(configuration.getRegion().asAws())
             .serviceConfiguration(pathStyleAccess)
+            .overrideConfiguration(builder -> builder.addMetricPublisher(new JamesS3MetricPublisher(metricFactory)))
```

Comment:
Configuration parameter (or system property) to enable/disable this just in case this is a bad idea.